### DISCUSSION TITLE
`experiments.outputModule` build broken code with `externals` options

### DISCUSSION THREAD

**Discussion by shoonia** [webpack/webpack#17292] [Q&A] [answered]
# Bug report

I want to exclude a transitive polyfill dependency from my build. I try to use the `externals` config by returning an object with the empty expected method.

```js
// my webpack.config.js
module.exports = {
  //...
  externals: {
    'react-lifecycles-compat': '{polyfill(){}}',
  },
  // ...
  experiments: {
    backCompat: false,
    outputModule: true,
  },
};
```

When the `experiments.outputModule` is not turn on, I have a valid build code. If I try to on `experiments.outputModule` (`true`), then webpack generate broken code like this one:

```js
import * as __WEBPACK_EXTERNAL_MODULE__polyfill__ae4b39ad__ from "{polyfill(){}}";
```

And I see en error in the browser console:

```console
Uncaught TypeError: Failed to resolve module specifier "{polyfill(){}}". Relative references must start with either "/", "./", or "../".
```

**Other relevant information:**
webpack version:  `^5.84.1`
Node.js version: `v16.14.0`
Operating System: `MacOS`
Additional tools:

**Comment by alexander-akait** [comment, marked as answer]
When you set `experiments.outputModule`, you need specify a type of your external, by default it will use `import` so you need to use `data:`, in commonjs you can have `const foo = require("my-external")`, so webpack replace it on `const foo = {polyfill(){}};`, but you can't use it in ESM, even more we strongly recommend to use path to module, not inline code, because it is very limited and works only in some cases, for example using `externalsType` (https://webpack.js.org/configuration/externals/#externalstype) with other values will be broken too (not for all, but for many)

**Reply by alexander-akait** [reply]
Also you can use this `'react-lifecycles-compat': 'assign {polyfill(){}}',` to fix it

**Comment by shoonia** [comment]
I didn't know this before, that `externals` has so rich API. I have fixed my issue with `externalsType`. Thank you, for help.
